### DISCUSSION TITLE
Version 1.0.0

### DISCUSSION THREAD

**Discussion by MattBrittan** [eclipse/paho.golang#239] [General]
Hi,

With the release of 0.20 I think we can now begin considering what is needed for a V1 release. I believe that the library now provides functionality to satisfy the majority of use-cases and is relatively easy to use. However there are some rough edges and a few features I feel are needed prior to V1 including:

- Async publish; notification that message has been delivered (or queued and connection closing)
- Manual ACK; consider how reconnections should be dealt with (potentially waiting for handlers to complete before reconnecting)
- Better docs

Really keen to hear your experiences with the library; what works well, what do we need to add and what should be changed? It's important that we try to complete any API changes now because this will become a lot more difficult once we pass V1.

I have created a [milestone](https://github.com/eclipse/paho.golang/milestone/1) and linked some issues I feel need to be resolved but thought this area (newly activated) would be the best place for a discussion!

Thanks,
Matt

**Comment by izarraga** [comment]
Hi,

I have release 0.20 working in production without problem.  Thanks for the work.

I think It would be nice to export length of the queue.

Regards,

Igor

**Reply by MattBrittan** [reply]
>I think It would be nice to export length of the queue.

This might be better as a feature request (in issues). I would not be keen to add this to the `interface` because I can see that it would be difficult/costly to calculate with some stores (the file-queue does not maintain this info and would need to scan the folder to get it).  Could easily be added to the Memory queue as it currently stands but not sure how easy it would be if PR #191 is accepted...
